### PR TITLE
Feat: Playtime & runtime

### LINKED ISSUE #11: Feat: Add playtime attribute to UserGames and runtime to Movies and Albums
Users should be able to add the amount of hours played to their game items. This playtime should then be shown in the user's collection and a cumulative counter should be shown in the statistics below the amount of games.

Movies and Albums should have a runtime attribute which details the length of the item in minutes. If the user has the item in their collection and its progress is set to "completed", the runtime should be added to a total hours watched/listened counter statistic below the amount of movies/albums.

### PR DESCRIPTION
**Pull request by m-m-mic**
Hey there,

this PR adds the ability for users to enter for how many hours they have played a game in their collection. Additionally movies & albums now have a runtime attribute. The total playtime of all games in the user's collection as well as the total runtime of all completed movies and albums are now displayed in the dashboard statistics. The playtime of games are also now displayed in the games table.

**The following components were added/changed**:
- Album, Movies and Games statistics - Now feature statistics related to the total playtime / runtime of the user's collection
- Media-table - New playtime column for games displays hours played
- Movie and Album page - Now display runtime of item if it exists
- Games page - Users can now enter for how many hours they've played the game
- Interfaces - Deleted unused row interfaces, added time to media interfaces

The GIF above shows the newly added "hours played", "hours watched" and "hours listened" statistics on the dashboard page.

The GIF above shows me changing the value of a game's playtime.

___

Thanks for the review!

### CODE REVIEW

File: src/app/core/components/game-statistics/game-statistics.component.html [hunk 1]
```diff
@@ -5,9 +5,17 @@
...
+      {{ statistics.amountInCollection | number: '1.0-0' : 'en-US' }}
       {{ statistics.amountInCollection === 1 ? 'Game' : 'Games' }}
     </span>
+    <span class="total-time">
+      {{ playtimeInHours | number: '1.0-0' : 'en-US' }}
+      {{ playtimeInHours === 1 ? 'hour played' : 'hours played' }}
+    </span>
+    <span>
+      {{ hoursPerEntry | number: '1.0-0' : 'en-US' }}
+      {{ hoursPerEntry === 1 ? 'hour per Game' : 'hours per Game' }}
```
Comment (georgiee) [inline src/app/core/components/game-statistics/game-statistics.component.html:17]: Maybe time now involve [ngPlural](https://angular.io/api/common/NgPlural) 😬

File: src/app/core/components/game-statistics/game-statistics.component.html [hunk 2]
```diff
@@ -5,9 +5,17 @@
   </div>
   <span class="stat-container main">
     <span class="total-amount">
-      {{ statistics.amountInCollection }}
+      {{ statistics.amountInCollection | number: '1.0-0' : 'en-US' }}
```
Comment (georgiee) [inline src/app/core/components/game-statistics/game-statistics.component.html]: en-US should be the default local right now. Why did you add it ?

File: src/app/core/components/game-statistics/game-statistics.component.ts [hunk 3]
```diff
@@ -13,4 +13,20 @@ import { Statistics } from '../../models/statistics.interface';
...
+
+  get playtimeInHours() {
+    if (!this.statistics) {
+      return 0;
+    }
+    return Math.round(this.statistics.time / 60);
+  }
+
+  get hoursPerEntry() {
+    if (!this.statistics || this.playtimeInHours === 0) {
```
Comment (georgiee) [inline src/app/core/components/game-statistics/game-statistics.component.ts:25]: `playtimeInHours` of 0 is perfectly fine the the calculation further down. I would not exit because of that. feels more special than it it. the statistics blocker is fine though

File: src/app/core/components/movie-statistics/movie-statistics.component.html [hunk 4]
```diff
@@ -5,9 +5,17 @@
   </div>
   <span class="stat-container main">
     <span class="total-amount">
-      {{ statistics.amountInCollection }}
+      {{ statistics.amountInCollection | number: '1.0-0' : 'en-US' }}
       {{ statistics.amountInCollection === 1 ? 'Movie' : 'Movies' }}
     </span>
+    <span class="total-time">
+      {{ runtimeInHours | number: '1.0-0' : 'en-US' }}
+      {{ runtimeInHours === 1 ? 'hour watched' : 'hours watched' }}
```
Comment (georgiee) [inline src/app/core/components/movie-statistics/movie-statistics.component.html:13]: time to think about a generalization because you're repeating code quite often is a very predictive way.
Maybe not generalize everything but a few components shared across the different features?

File: src/app/core/services/users.service.ts [hunk 5]
```diff
@@ -86,6 +85,7 @@ export class UsersService {
             platform: gameData.platform ?? '',
             format: convertFormat(game.format),
             progress: convertProgress(game.progress),
+            time: game.playtime ?? 0,
```
Comment (georgiee) [inline src/app/core/services/users.service.ts:88]: 👍

**Review by georgiee — APPROVED** [review]
Hey that feature is a nice idea! The code to implement it looks good and understandable. Code repetition happens as usually so maybe start thinking about how to reuse smaller parts.

Otherwise good 👌 Let's merge

File: src/app/core/components/game-statistics/game-statistics.component.html (hunk 2 above)
Comment (m-m-mic) [reply src/app/core/components/game-statistics/game-statistics.component.html]: Wasn't sure if angular defaults to en-US or bases it on browser language so I wanted to be on the safe side

File: src/app/core/components/game-statistics/game-statistics.component.ts (hunk 3 above)
Comment (m-m-mic) [reply src/app/core/components/game-statistics/game-statistics.component.ts:25]: Screenshot 2023-10-12 at 15 50 06

I have to catch it or else the 0 doesn't show up if the user has no items in their collection.
